modeshape-jcr/src/main/resources/org/modeshape/jcr/modeshape_builtins.cnd

```diff
@@ -140,6 +140,22 @@
     < 'STRING', 'URI', 'BINARY', 'LONG', 'DOUBLE', 'DECIMAL', 'BOOLEAN', 
       'DATE', 'NAME', 'PATH', 'REFERENCE', 'WEAKREFERENCE', 'UNDEFINED'
 
+// -----------------------
+// Large unordered collections
+// -----------------------
+// Abstract base type for all unordered collections.
+
+[mode:unorderedCollection] abstract
+  + * (nt:base) = nt:unstructured abort 
```

Comment:
>Anyway, isn't it possible to change the primary type of an item?

good point, I missed that. Then that's an extra validation that should be there - i.e. you should not be able to change the node type to/from such a collection.

>What if we were to require the primary type have mode:unorderedCollection as a base? That might even simplify the code a bit, since we'd never have to check mixins.

not sure I follow: we don't test mixins right now and since `mode:unorderedCollection` is abstract, one of the 4 subtypes should be used - which also defines the number of buckets.